sky/packages/sky/lib/src/widgets/scrollable.dart

```diff
@@ -54,15 +55,25 @@ abstract class Scrollable extends StatefulComponent {
 abstract class ScrollableState<T extends Scrollable> extends State<T> {
   void initState() {
     super.initState();
-    if (config.initialScrollOffset is double)
-      _scrollOffset = config.initialScrollOffset;
     _animation = new SimulationStepper(_setScrollOffset);
+    if (config.key != null) {
+      _route = Route.of(context);
+      if (_route != null) {
+        double lastOffset = _route.readState(config.key);
+        if (lastOffset != null)
+          _scrollOffset = lastOffset;
+      }
+    }
+    if (_scrollOffset == null && config.initialScrollOffset != null)
+      _scrollOffset = config.initialScrollOffset;
   }
 
+  Route _route;
```

Comment:
Caching the route isn't strictly correct because this state object can be moved to another subtree (potentially with a different route) if one of its ancestors has a global key.